src/xfinity.ts

```diff
@@ -216,7 +241,8 @@ export class Xfinity extends EventEmitter {
         if (this.#page?.isClosed() !== false) {
             const browser = await this.getBrowser();
             const page = await browser.newPage();
-
+            
+            this.#Cookies.readCookies().map(async c => await page.setCookie(c));
```

Comment:
```suggestion
            await page.setCookie(...this.#Cookies.readCookies());
```

Reply:
Great suggestion. Cleaned up the code in my original commit.